docs/pdk_generating_modules.md

```diff
@@ -0,0 +1,84 @@
+* Module version. We use and recommend semantic versioning for modules.
+* The module author's name.
+* The license under which your module is made available. This should be an identifier from [SPDX License List](https://spdx.org/licenses/).
+* A one-sentence summary about your module.
+* The URL to your module's source code repository, so that other users can contribute back to your module.
+* The URL to a web site that offers full information about your module, if you have one..
+* The URL to the public bug tracker for your module, if you have one.
+
+After you generate a module, we suggest validating and testing the module _before_ you add classes or write new code in it. This allows you to verify that the module files and directories were correctly created.
+
+PDK does not generate any classes at module creation. The `pdk new class` command creates a class manifest and a test template file for the class. When you run this command, PDK creates a class manifest and a test template file for the class. You can then write tests in this template to validate your class's behavior.
+
+If your new class should take parameters, you can specify them, along with the parameter's data type and values, on the command line when you generate your class. You can provide any number of parameters on the command line.
+
+
+## Generate a module with pdk
+
+To generate a module with PDK's default template, use the `pdk new module` command.
+
+Before you begin, ensure that you've installed the PDK package.
+
+1. From the command line, run the `pdk new module` command, specifying the name of the module: `pdk new module module_name`
+   
+   Optionally, to skip the interview questions and generate the module with default values, use the ``skip-interview` flag when you generate the module: `pdk new module module_name --skip-interview`
+
+1. Respond to the PDK dialog questions in the terminal. Each question indicates the default value it will use if you just hit **Enter**.
+
+   1. Forge username: Enter your Forge username, if you have a Forge account.
+   2. Version: Enter the semantic version of your module, such as "0.1.0".
+   3. Author: Enter the name of the module author.
+   4. License: If you want to specify a license other than "Apache-2.0," specify that here, such as "MIT", or "proprietary".
+   5. Description: Enter a one-sentence summary that helps other users understand what your module does.
+   6. Source code repository: Enter the URL to your module's source code repository.
+   7. Where others can learn more: If you have a website where users can learn more about your module, enter the URL.
+   8. Where others can report issues: If you have a public bug tracker for your module, enter the URL.
+
+1. If the metadata that PDK displays is correct, confirm to generate the module. If it is incorrect, enter `n` to cancel and start over.
```

Comment:
I was thinking about this, but it also says what to do right in the interface. That said, I can add it though, I think it seems OK with it too.